src/Entity/Geo/Zone.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace App\Entity\Geo;
+
+use Algolia\AlgoliaSearchBundle\Mapping\Annotation as Algolia;
+use App\Entity\EntityTimestampableTrait;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity(repositoryClass="App\Repository\Geo\ZoneRepository")
+ * @ORM\Table(
+ *     name="geo_zone",
+ *     uniqueConstraints={
+ *         @ORM\UniqueConstraint(name="elected_representative_zone_name_category_unique", columns={"code", "type"})
+ *     }
+ * )
+ * @ORM\AttributeOverrides({
+ *     @ORM\AttributeOverride(name="code", column=@ORM\Column(unique=false))
+ * })
+ *
+ * @Algolia\Index(autoIndex=false)
+ */
+class Zone implements GeoInterface
+{
+    use GeoTrait;
+    use EntityTimestampableTrait;
+
+    public const COUNTRY = 'country';
+    public const REGION = 'region';
+    public const DEPARTMENT = 'department';
+    public const DISTRICT = 'district';
+    public const CITY = 'city';
+    public const CITY_COMMUNITY = 'city_community';
+    public const CANTON = 'canton';
+    public const FOREIGN_DISTRICT = 'foreign_district';
+    public const CONSULAR_DISTRICT = 'consular_district';
+
+    /**
+     * @var int|null
+     *
+     * @ORM\Id
+     * @ORM\Column(type="integer")
+     * @ORM\GeneratedValue
+     */
+    private $id;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(nullable=false)
+     */
+    private $type;
+
+    /**
+     * @var Collection
+     *
+     * @ORM\ManyToMany(targetEntity="App\Entity\Geo\Zone", inversedBy="children")
+     * @ORM\JoinTable(
+     *     name="geo_zone_parent",
+     *     joinColumns={@ORM\JoinColumn(name="child_id", referencedColumnName="id")},
+     *     inverseJoinColumns={@ORM\JoinColumn(name="parent_id", referencedColumnName="id")}
+     * )
+     */
+    private $parents;
+
+    /**
+     * @var Collection
+     *
+     * @ORM\ManyToMany(targetEntity="App\Entity\Geo\Zone", mappedBy="parents")
+     */
+    private $children;
+
+    public function __construct(string $type, string $code, string $name)
+    {
+        $this->type = $type;
+        $this->code = $code;
+        $this->name = $name;
+        $this->parents = new ArrayCollection();
+        $this->children = new ArrayCollection();
+    }
+
+    public function __toString(): string
```

Comment:
same for this method, which added by GeoTrait

Reply:
fixed